client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -151,7 +150,11 @@ const Checkout = React.createClass( {
 			return false;
 		}
 
-		if ( this.props.transaction.step.name === transactionStepTypes.SUBMITTING_WPCOM_REQUEST ) {
+		// Do not redirect if cart is empty, but we're submitting or received transaction response
+		if ( includes(
+			[ transactionStepTypes.SUBMITTING_WPCOM_REQUEST, transactionStepTypes.RECEIVED_WPCOM_RESPONSE ],
```

Comment:
Is the step set to `RECEIVED_WPCOM_RESPONSE` if there is an error? If so, I think this will break exiting checkout by emptying cart once there has been an error.